includes/blocks/class-sensei-course-navigation-block.php

```diff
@@ -0,0 +1,294 @@
+					</button>';
+		}
+
+		return '
+			<section ' . Sensei_Block_Helpers::render_style_attributes( $classes, [] ) . '>
+				<header class="sensei-lms-course-navigation-module__header">
+					<h2 class="sensei-lms-course-navigation-module__title">
+						<a href="' . esc_url( $module_url ) . '">' . $title . '</a>
+					</h2>
+					' . $collapse_toggle .
+			'</header>
+				<div class="sensei-lms-course-navigation-module__lessons sensei-collapsible__content ' . $collapsed . '">
+					' . $lessons_html . '
+				</div>
+				<div class="sensei-lms-course-navigation-module__summary">
+				' . wp_kses_post( $summary ) . '
+				</div>
+			</section>
+		';
+	}
+
+	/**
+	 * Build lesson HTML.
+	 *
+	 * @param array $lesson Lesson data.
+	 *
+	 * @return string
+	 */
+	public function render_lesson( $lesson ) {
+		$lesson_id  = $lesson['id'];
+		$status     = $this->get_user_lesson_status( $lesson_id );
+		$is_current = Sensei_Utils::get_current_lesson() === $lesson_id;
+		$has_quiz   = Sensei_Lesson::lesson_quiz_has_questions( $lesson_id );
+		$quiz_id    = Sensei()->lesson->lesson_quizzes( $lesson_id );
+
+		$classes = [ 'sensei-lms-course-navigation-lesson', 'status-' . $status ];
+
+		if ( $is_current ) {
+			$classes[] = 'current-lesson';
+		}
+
+		$lesson_quiz_html = '';
+
+		if ( $has_quiz ) {
+			$lesson_quiz_html = '<a class="sensei-lms-course-navigation-lesson__quiz" href="' . esc_url( get_permalink( $quiz_id ) ) . '">' . esc_html__( 'Quiz', 'sensei-lms' ) . '</a>';
+		}
+
+		return '
+		<div ' . Sensei_Block_Helpers::render_style_attributes( $classes, [] ) . '>
+			<a href="' . esc_url( get_permalink( $lesson_id ) ) . '" class="sensei-lms-course-navigation-lesson__link">
+				' . self::lesson_status_icon( $status ) . '
+				<span class="sensei-lms-course-navigation-lesson__title">
+					' . esc_html( $lesson['title'] ) . '
+				</span>
+			</a>
+			' . $lesson_quiz_html . '
+		</div>';
+	}
+
+	/**
+	 * Get the lesson status icon.
+	 *
+	 * @param string $status
+	 * @return string Icon HTML.
+	 */
+	public static function lesson_status_icon( $status ) {
+		return '<svg class="sensei-lms-course-navigation-lesson__status">
+					<use xlink:href="#sensei-lesson-status-' . $status . '"></use>
+				</svg>';
+	}
+
+	/**
+	 * Get the lesson status string for the user.
+	 *
+	 * @param int $lesson_id
+	 *
+	 * @return string
+	 */
+	private function get_user_lesson_status( $lesson_id ): string {
+		$status    = 'not-started';
+		$completed = Sensei_Utils::user_completed_lesson( $lesson_id, get_current_user_id() );
+
+		if ( $completed ) {
+			$status = 'completed';
+		} else {
+			$user_lesson_status = Sensei_Utils::user_lesson_status( $lesson_id, get_current_user_id() );
+			if ( isset( $user_lesson_status->comment_approved ) ) {
+				$status = $user_lesson_status->comment_approved;
+			}
+		}
+		return $status;
+	}
+
+	/**
+	 * Build HTML to reference SVG icons from.
+	 *
+	 * @return string
+	 */
+	private function render_svg_icon_library() {
```

Comment:
After merging the open PRs it would be nice to unify the SVGs approach. I created a draft issue for that as a reminder. :)